thunder/tests/test_nvfuser.py

```diff
@@ -974,34 +974,38 @@ def foo(b):
     dtypes=NOTHING,
 )
 def test_scatter(executor, device: str, dtype: dtypes.dtype):
-    def foo(inputs: list):
-        inp: torch.Tensor
-        idxs: torch.Tensor
-        src: torch.Tensor
-        inp, idxs, src = inputs
-        out = torch.scatter(inp, 0, idxs, src)
-        return out
-
     bsz = 4
     hidden = 8
     scatter_size = 3
     x = torch.randn([bsz, hidden], device=device, dtype=dtype)
-    _, ind = torch.topk(x, k=scatter_size, dim=0)
-    src = torch.randn(scatter_size, hidden, device=device, dtype=dtype)
 
-    # NOTE nv_enable_scatter to allow scatter operation to go through nvfuser
-    jfoo = thunder.jit(foo, nv_enable_scatter=True)
+    for dim in (0, 1):
 
-    inputs = [x, ind, src]
+        def foo(inputs: list):
+            inp: torch.Tensor
+            idxs: torch.Tensor
+            src: torch.Tensor
+            axis: int
+            inp, idxs, src, axis = inputs
+            out = torch.scatter(inp, axis, idxs, src)
+            return out
 
-    actual = jfoo(inputs)
-    expected = foo(inputs)
-    torch.testing.assert_close(actual, expected)
+        _, ind = torch.topk(x, k=scatter_size, dim=dim)
+        src = torch.randn(ind.shape, device=device, dtype=dtype)
 
-    fw_trace = thunder.last_traces(jfoo)[-1]
-    fusion_bsyms = tuple(filter(lambda a: a.sym.is_fusion, fw_trace.bound_symbols))
+        # NOTE nv_enable_scatter to allow scatter operation to go through nvfuser
+        jfoo = thunder.jit(foo, nv_enable_scatter=True)
 
-    # assert that everything is merged as a single nvfuser operation
-    assert len(fusion_bsyms) == 1
-    outside_fusion_syms = ["unpack_trivial", "python_return"]
-    assert {el.sym.name for el in fw_trace.bound_symbols if not el.sym.is_fusion} == set(outside_fusion_syms)
+        inputs = [x, ind, src, dim]
```

Comment:
I think the inputs are in the order of `fd.ops.scatter`'s parameters but I think it would be more readable to have them in same order as `torch.scatter`. That way we can also directly do `thunder.jit(torch.scatter, nv_enable_scatter=True)` and get rid of `foo`.